### ISSUE TITLE
Clarification on validation split in train_align.py

### ISSUE THREAD

**Issue by R2covery** [kkkkk001/PoinnCARE#1]
Hello,

I am reproducing the experiments of PoinnCARE on the CARE benchmark and noticed a possible discrepancy between the paper description and the released code.

The paper describes the four CARE datasets as test sets and states that, under the inductive setting, test enzymes are withheld until inference.

However, in the released code:

1. `utils/data_utils.py` assigns the validation split as follows:

```python
splits['val'] = splits['30_protein_test']
```

2. train_align.py evaluates val accuracy for checkpoint selection and early stopping:
val_metrics_align = model1.compute_metrics(
    embeddings1, data, 'val', output
)

3. When the validation accuracy improves, the code saves the checkpoint and evaluates all four test sets.
Therefore, the <30% Identity test set appears to be used both as the validation set and as one of the final test sets.

Could you please clarify:
1. Is using 30_protein_test as the validation set intentional?
2. Should users instead create an internal train/dev split from protein_train50.csv and reserve all four CARE datasets exclusively for final evaluation?

This clarification would be helpful for reproducing the reported results under a strictly held-out test protocol.
Thank you.
